slangpy/core/calldata.py

```
@@ -478,93 +480,52 @@ def build(self, build_info: "FunctionBuildInfo", *args: Any, **kwargs: Any):
+                    ta = binding.vector_type.access
+                    if ta == TensorAccess.read:
+                        access_list.append(AutogradAccess.read)
+                    elif ta == TensorAccess.write:
+                        access_list.append(AutogradAccess.write)
+                    elif ta == TensorAccess.read_write:
+                        access_list.append(AutogradAccess.readwrite)
+                    else:
+                        access_list.append(AutogradAccess.none)
+                else:
+                    a = binding.access[0]
+                    if a == AccessType.read:
```

Ruff (0.15.2)

[error] `AccessType` may be undefined, or defined from star imports

(F405)